octokit/search.go

```diff
@@ -2,47 +2,60 @@ package octokit
 
 import (
 	"github.com/jingweno/go-sawyer/hypermedia"
-	"net/url"
 )
 
-var SearchURL = Hyperlink("search{/type}?q={query}{&page,per_page,sort,order}")
+var SearchURITemplate = "search{/type}?q={query}{&page,per_page,sort,order}"
 
-func (c *Client) Search(url *url.URL) (searches *SearchService) {
-	searches = &SearchService{client: c, URL: url}
-	return
+func (c *Client) Search(uriTemplate string) *SearchService {
+	return &SearchService{&GenericService{client: c, uriTemplate: uriTemplate}}
 }
 
 // A service to return search records
 type SearchService struct {
-	client *Client
-	URL    *url.URL
+	*GenericService
```

Comment:
We _may_ end up here, but I'd like to see a couple of implementations before we extract this. @lalalah's work on Gists might help.